pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -387,6 +387,17 @@ func (c *Reconciler) resolvePipelineState(
 				return nil, controller.NewPermanentError(err)
 			}
 		}
+
+		if config.FromContextOrDefaults(ctx).FeatureFlags.EnableParamEnum {
+			if len(resolvedTask.TaskRuns) > 0 && len(resolvedTask.TaskRuns[0].Status.Conditions) > 0 {
+				cond := resolvedTask.TaskRuns[0].Status.Conditions[0]
```

Comment:
Is it possible that the Conditions[0] is not the condition we need? We can loop the conditions to fetch the apisucceed condition
And curious if we have a permanent error from taskrun, is it surfaced somewhere?

Reply:
Oh, good point. If a Matrix is used and multiple TaskRun is created with different combintations. We do need to validate all the results.

And yes, we do have a permanent error from TaskRun, which is also Propogated back to the owning `PipelineRun`:
https://github.com/tektoncd/pipeline/blob/7e3897a6353f81836e2f9186259bb78c21434b20/pkg/reconciler/taskrun/taskrun.go#L420C1-L426C3